### ISSUE TITLE
libSDL2-2.0.0.dylib missing on 10.15.5 macOS

### ISSUE THREAD

**Issue by tommyw1979** [projectM-visualizer/projectm#363]
Process:               SDLprojectM [21608]
Path:                  /Applications/projectM SDL.app/Contents/MacOS/SDLprojectM
Identifier:            net.projectm.projectM-SDL
Version:               3.1.3 (1)
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           SDLprojectM [21608]
User ID:               501

Date/Time:             2020-05-30 03:14:46.660 +0100
OS Version:            Mac OS X 10.15.5 (19F96)
Report Version:        12
Anonymous UUID:        30DB279A-A1A9-4D97-A3F6-14EEDAD8F74A

Time Awake Since Boot: 39000 seconds

System Integrity Protection: disabled

Crashed Thread:        0

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    DYLD, [0x1] Library missing

Application Specific Information:
dyld: launch, loading dependent libraries

Dyld Error Message:
  Library not loaded: /usr/local/opt/sdl2/lib/libSDL2-2.0.0.dylib
  Referenced from: /Applications/projectM SDL.app/Contents/MacOS/SDLprojectM
  Reason: image not found

Binary Images:
       0x102111000 -        0x1021a8ff7 +net.projectm.projectM-SDL (3.1.3 - 1) <F62706AE-3642-39A3-A7E4-B44CD03599E2> /Applications/projectM SDL.app/Contents/MacOS/SDLprojectM
       0x107687000 -        0x107718eff  dyld (750.5) <26346F4C-B18E-31A1-9964-30736214F1BF> /usr/lib/dyld

Model: iMacPro1,1, BootROM 1037.100.359.0.0, 4 processors, Quad-Core Intel Core i7, 4.01 GHz, 16 GB, SMC 1.74f4
Graphics: Radeon RX 560, Radeon RX 560, spdisplays_pcie_device, 4 GB
Memory Module: BANK 0/DIMM0, 4 GB, DDR4, 2400 MHz, Crucial Technology, BLS4G4D240FSA.8FARG
Memory Module: BANK 1/DIMM0, 4 GB, DDR4, 2400 MHz, Crucial Technology, BLS4G4D240FSA.8FARG
Memory Module: BANK 0/DIMM1, 4 GB, DDR4, 2400 MHz, Crucial Technology, BLS4G4D240FSA.8FARG
Memory Module: BANK 1/DIMM1, 4 GB, DDR4, 2400 MHz, Crucial Technology, BLS4G4D240FSA.8FARG
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0x111), Broadcom BCM43xx 1.0 (7.77.111.1 AirPortDriverBrcmNIC-1610.1)
Bluetooth: Version 7.0.5f6, 3 services, 27 devices, 1 incoming serial ports
Network Service: Wi-Fi, AirPort, en1
PCI Card: Intel Corporation, Series Chipset USB xHCI Host Controller, USB Controller, Built In
PCI Card: Intel Corporation, Series Chipset MEI Controller, IMEI Controller, Built In
PCI Card: Intel Corporation, Series Chipset SATA Controller, AHCI Controller, Built In
PCI Card: Patched by MaLd0n - olarila.com, MaLd0n, Built In
PCI Card: Radeon RX 560, sppci_displaycontroller, PCI Slot 0
PCI Card: Apple HDMI Audio Controller, HDMI Controller, Built In
PCI Card: Intel Corporation, Series Chipset PCHPMC Controller, PCHPMC Controller, Built In
PCI Card: Intel I219V2 PCI Express Gigabit Ethernet, Ethernet Controller, Built In
PCI Card: Broadcom Brcm43xx Wireless Network Adapter, AirPort Extreme, Built In
Serial ATA Device: TOSHIBA Q300., 480.1 GB
Serial ATA Device: ST4000DM004-2CV104, 4 TB
Serial ATA Device: HL-DT-ST DVDRAM GH24NS95
Serial ATA Device: SanDisk SDSSDH3 4T00, 4 TB
Serial ATA Device: Samsung SSD 850 EVO 500GB, 500.11 GB
Serial ATA Device: Hitachi HDS5C3030BLE630, 3 TB
USB Device: USB 3.1 Bus
USB Device: Hub
USB Device: USB Controls
USB Device: USB 3.0 Bus
USB Device: Backup+  Desk
USB Device: USB Receiver
USB Device: USB2.0 Hub
USB Device: BRCM20702 Hub
USB Device: Bluetooth USB Host Controller
USB Device: UMC204HD 192k
Thunderbolt Bus:

**Comment by milkdropper** [comment]
Same issue I saw here https://github.com/projectM-visualizer/projectm/issues/353 (I'll close that one, since you gave more details). 

For now 3.1.2 seems to work. https://github.com/projectM-visualizer/projectm/releases/tag/v3.1.2

**Comment by revmischa** [comment]
Looks like an issue with packaging SDL2.framework

**Comment by revmischa** [comment]
Can you please test if this fixes the issue?

[projectM-macOS.pkg.zip](https://github.com/projectM-visualizer/projectm/files/4746284/projectM-macOS.pkg.zip)

**Comment by milkdropper** [comment]
No it's the same error:
Library not loaded: /usr/local/opt/sdl2/lib/libSDL2-2.0.0.dylib

I noticed that the only change between 3.1.2 and 3.1.3 is input toggling, but the file size reduced by more than a megabyte. Also the zip you just sent is small like the 3.1.3 release. Are you compiling it without packaging SDL inside the binary? I am only guessing.

**Comment by milkdropper** [comment]
So I am trying to get a new build to work on a "fresh" macOS install (that does not have SDL2 libraries installed). It seems it's related to /Frameworks/SDL2.framework being in the 3.1.2 package but not in 3.1.3.

I am not familiar with xcode and building on macOS. @revmischa can you help? I tried this http://matthewstyles.com/set-up-an-sdl2-project-in-xcode/ and see the framework being copied into my build/debug directories but not into the app package.

**Comment by revmischa** [comment]
I am not super sure why. My understanding is that libsdl2 should be being statically linked but apparently that isn't happening or it's also expecting the framework.
Issue may be related to these XCode changes - https://github.com/projectM-visualizer/projectm/compare/v3.1.2...master#diff-5c44692eccb37aa51cf9643ca5bd06edL124

**Comment by milkdropper** [comment]
I still couldn't figure this out. Can you make sure you linked to the right diff? All I see when I follow that link is "There isn’t anything to compare."

**Comment by revmischa** [comment]
Recent changes to the pbxproj: https://github.com/projectM-visualizer/projectm/compare/v3.1.2...v3.1.3#diff-5c44692eccb37aa51cf9643ca5bd06ed

**Comment by milkdropper** [comment]
I believe I got static linking of libSDL working on macOS. Is anyone able to test by installing one of these keys in Steam?

QWQIT-K8P90-9ZK48
QW7QZ-8GLXB-II5HR
C83P7-BHTL4-9YFCG

**Comment by revmischa** [comment]
I tried the first steam code, was able to run it. I don't have SDL2 installed via brew or /Library/Frameworks.
First time I ran it I had a black screen, when I force quit I was able to see a dialog asking for microphone access that was behind the black screen.

<img width="617" alt="Screen Shot 2020-07-25 at 5 54 34 PM" src="https://user-images.githubusercontent.com/245131/88459747-4838e980-cea0-11ea-830b-8cd9d3846513.png">

It seems very slow. Is the steam binary an optimized build?

**Comment by milkdropper** [comment]
I also noticed the microphone request that popped up in the background on first run.  I think macOS is getting more and more strict about security. The biggest problem with this release is that I built it in Catalina and it won't run on an older OS. I think this attempt at compiling without xcode is not ideal, because it is much easier to set the minimum OS in xcode. 

Since I could not figure out static linking in xcode, my next idea is to followed by using instasll_name_tool to link to the relative framework path instead of an absolute path.
